### DISCUSSION TITLE
Object '""' does not exist or not authorized.

### DISCUSSION THREAD

**Discussion by BFASRachelS** [sqlalchemy/sqlalchemy#11364] [Usage Questions] [answered]
I am new to sqlalchemy and I am having an issue querying Snowflake. I get this error.

sqlalchemy.exc.ProgrammingError: (snowflake.connector.errors.ProgrammingError) 002003 (42S02): SQL compilation error:
Object '"Post"' does not exist or not authorized.
[SQL: SELECT "Post"."ID" AS "Post_ID", "Post"."TITLE" AS "Post_TITLE", "Post"."CONTENT" AS "Post_CONTENT" 
FROM "Post"]

If I run the SQL above directly in Snowflake it gives me the same error. Seems like this might be something to do with the query syntax being generated. Surely this should be generated without the quotes. For example.

SELECT Post.ID AS "Post_ID", Post.TITLE AS "Post_TITLE", Post.CONTENT AS "Post_CONTENT" 
FROM Post

This is my code. It's a very simple example so, if someone could educate me, I'd be extremely grateful.

```
engine = create_engine(
    url=URL(
                account="",
                warehouse="",
                database="",
                schema="",
                user="",
                password=""
        ),
)


# Create a SQLALCHEMY model definition for the Post table
class Post(Base):
    """Class that represents a Post"""

    __tablename__: str = "Post"
    ID = Column(Integer, primary_key=True)
    TITLE = Column(String(100))
    CONTENT = Column(String(100))

    def __init__(self, ID, TITLE, CONTENT):
        self.ID = ID
        self.TITLE = TITLE
        self.content = CONTENT


Session = sessionmaker(bind=engine)
session1 = Session()



posts = session1.query(Post).all()
print('\n### All Posts:')
for post in posts:
    print(f'{post.ID} has {post.TITLE}')
print('')


```

**Comment by zzzeek** [comment, marked as answer]
hi -

if you dont want quotes, use [all lowercase names](https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.Column.params.name):

```py
class Post(Base):
    """Class that represents a Post"""

    __tablename__: str = "post"
    id = Column(Integer, primary_key=True)
    title = Column(String(100))
    content = Column(String(100))
```


if you really want those attributes to be UPPERCASE, you can assign a separate name:

```py
class Post(Base):
    """Class that represents a Post"""

    __tablename__: str = "post"
    ID = Column("id", Integer, primary_key=True)
    TITLE = Column("title", String(100))
    CONTENT = Column("content", String(100))

```

**Reply by BFASRachelS** [reply]
Thanks. I will file that one under newbie stupidity. :-)

**Reply by zzzeek** [reply]
we dont document that part very well, it's buried.  no worries
